packages/stargate/src/stargateclient.ts

```diff
@@ -303,14 +300,17 @@ export class StargateClient {
             height: result.height,
             rawLog: result.rawLog,
             transactionHash: txId,
+            gasUsed: gasUsed,
           }
-        : pollForTx(txId);
+        : pollForTx(txId, gasUsed);
     };
 
     return new Promise((resolve, reject) =>
       this.forceGetTmClient()
         .broadcastTxSync({ tx })
-        .then(({ hash }: BroadcastTxSyncResponse) => pollForTx(toHex(hash).toUpperCase()))
+        .then(({ gasUsed, hash }: tendermint34.BroadcastTxSyncResponse) =>
+          pollForTx(toHex(hash).toUpperCase(), gasUsed),
```

Comment:
Why do we need to pass the value here from the `sync` response? How can `sync` broadcast even know the value?

Can't we get it from `const result = await this.getTx(txId);`?